### PR TITLE
remove base amounts from base strategy

### PR DESCRIPTION
**Pull request by belbix**

### CODE REVIEW

File: contracts/strategy/StrategyBaseV2.sol [hunk 1]
```diff
@@ -141,7 +149,6 @@ abstract contract StrategyBaseV2 is IStrategyV2, ControllableV3 {
     address _asset = asset;
     // gas saving
     uint balance = IERC20(_asset).balanceOf(address(this));
-    _decreaseBaseAmount(_asset, baseAmounts[_asset]);
     // reset base amount
```
Comment (dvpublic) [inline contracts/strategy/StrategyBaseV2.sol:152]: we don't need this comment anymore

File: contracts/test/MockStrategy.sol [hunk 2]
```diff
@@ -155,9 +152,9 @@ contract MockStrategy is StrategyBaseV2 {
     compoundRatio = ratio;
   }
 
-  function setBaseAmount(address asset_, uint amount_) external {
-    baseAmounts[asset_] = amount_;
-  }
+//  function setBaseAmount(address asset_, uint amount_) external {
```
Comment (dvpublic) [inline contracts/test/MockStrategy.sol:155]: we don't need it anymore

File: contracts/strategy/StrategyBaseV2.sol [hunk 3]
```diff
@@ -155,56 +162,43 @@ abstract contract StrategyBaseV2 is IStrategyV2, ControllableV3 {
...
+  /// @return strategyLoss Loss should be covered from Insurance
+  function withdrawAllToSplitter() external override returns (uint strategyLoss) {
     address _splitter = splitter;
     address _asset = asset;
     StrategyLib.onlySplitter(_splitter);
 
     uint balance = IERC20(_asset).balanceOf(address(this));
 
-    (uint investedAssetsUSD, uint assetPrice, int totalAssetsDelta) = _withdrawAllFromPool();
+    (uint investedAssetsUSD, uint assetPrice, uint _strategyLoss) = _withdrawAllFromPool();
```
Comment (dvpublic) [inline contracts/strategy/StrategyBaseV2.sol:201]: investedAssetsUSD => expectedWithdrewUSD
It worth to rename investedAssetsUSD in StrategyLib.checkWithdrawImpact too

File: contracts/strategy/StrategyBaseV2.sol [hunk 4]
```diff
@@ -246,7 +229,7 @@ abstract contract StrategyBaseV2 is IStrategyV2, ControllableV3 {
       uint investedAssetsUSD;
       uint assetPrice;
 
-      (investedAssetsUSD, assetPrice, totalAssetsDelta) = _withdrawFromPool(amount - balance);
+      (investedAssetsUSD, assetPrice, strategyLoss) = _withdrawFromPool(amount - balance);
```
Comment (dvpublic) [inline contracts/strategy/StrategyBaseV2.sol:232]: investedAssetsUSD => expectedWithdrewUSD

File: contracts/vault/StrategySplitterV2.sol [hunk 5]
```diff
@@ -348,37 +349,31 @@ contract StrategySplitterV2 is ControllableV3, ReentrancyGuard, ISplitter {
...
     // slippage for withdraw
-    if (balanceAfterWithdraw < balance) {
-      uint loss = balance - balanceAfterWithdraw;
-      ITetuVaultV2(vault).coverLoss(loss);
-      emit Loss(lowStrategy, loss);
-      slippage = loss * 100_000 / balance;
+    if (strategyLossOnWithdraw != 0) {
+      ITetuVaultV2(vault).coverLoss(strategyLossOnWithdraw);
+      emit Loss(lowStrategy, strategyLossOnWithdraw);
+      slippage = strategyLossOnWithdraw * 100_000 / balance;
```
Comment (dvpublic) [inline contracts/vault/StrategySplitterV2.sol:369]: Now strategyLossOnWithdraw can include loss related to price-changes. So, in the case of high price change you will get high value of loss and most probably too-high slippage. The slippage shouldn't depend on price changes (?) There is no such problem with strategyLossOnInvest below because strategyLossOnInvest doesn't include losses related to price changes.

File: contracts/vault/StrategySplitterV2.sol [hunk 6]
```diff
@@ -538,12 +511,10 @@ contract StrategySplitterV2 is ControllableV3, ReentrancyGuard, ISplitter {
 
         remainingAmount = withdrew <= remainingAmount ? remainingAmount - withdrew : 0;
```
Comment (dvpublic) [inline contracts/vault/StrategySplitterV2.sol:512]: withdrew < remainingAmount

File: contracts/vault/StrategySplitterV2.sol [hunk 7]
```diff
@@ -724,13 +677,12 @@ contract StrategySplitterV2 is ControllableV3, ReentrancyGuard, ISplitter {
   /// @param updateTotalAssetsBeforeInvest TotalAssets of strategy should be updated before investing.
   ///                                      The increment of {TotalAssets} should be returned as {totalAssetsDelta}
```
Comment (dvpublic) [inline contracts/vault/StrategySplitterV2.sol:678]: second line of the commment should be deleted

File: contracts/vault/StrategySplitterV2.sol (hunk 5 above)
Comment (belbix) [reply contracts/vault/StrategySplitterV2.sol:369]: its fine
probably for invest need to check too
